zebra/zebra_rib.c

```diff
@@ -2232,10 +2231,34 @@ void _route_entry_dump(const char *func, union prefixconstptr pp,
 		   re->nexthop_num, re->nexthop_active_num);
 
 	for (ALL_NEXTHOPS(re->ng, nexthop)) {
-		inet_ntop(p->family, &nexthop->gate, straddr, INET6_ADDRSTRLEN);
-		zlog_debug("%s: %s %s[%u] vrf %u with flags %s%s%s", func,
+		struct vrf *vrf = vrf_lookup_by_id(nexthop->vrf_id);
+		int af = AF_INET;
+
+		switch (nexthop->type) {
+			/* It is possible that we want to treat blackhole
+			 * and ifindex differently than a v4 address,
+			 * not sure what we would do differently though
```

Comment:
I think you can fill the `straddr` differently depending on the nexthop type. Example: print the interface name for `NEXTHOP_TYPE_IFINDEX`, print "blackhole" for `NEXTHOP_TYPE_BLACKHOLE`, print the nexthop address and interface name for `NEXTHOP_TYPE_IPV4_IFINDEX`, etc.

Reply:
fixed